src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,720 @@
+                except Exception as ex:
+                    logging.error("Got unhandled exception while setting priority for job %i\n", job['id'])
+                    logging.error(str(ex))
+                    logging.error("Not setting priority")
+
+            jdl.append("priority = %i\n" % (task_priority + prio * self.maxTaskPriority))
+
+            jdl.append("+PostJobPrio1 = -%d\n" % len(job.get('potentialSites', [])))
+            jdl.append("+PostJobPrio2 = -%d\n" % job['taskID'])
+
+            jdl.append("+WMAgent_JobID = %s\n" % job['jobid'])
+
+            jdl.append("job_machine_attrs = GLIDEIN_CMSSite\n")
+
+            ### print all the variables needed for us to rely on condor userlog
+            jdl.append("job_ad_information_attrs = JobStatus,QDate,EnteredCurrentStatus,JobStartDate,DESIRED_Sites,ExtDESIRED_Sites,WMAgent_JobID,MATCH_EXP_JOBGLIDEIN_CMSSite\n")
+
+            jdl.append("Queue 1\n")
+
+        return jdl
+
+    def customizePerJob(self, job):
+        """
+        JDL additions just for this implementation. Over-ridden in sub-classes
+        These are the Glide-in specific bits
+        """
+        jdl = []
+        jobCE = job['location']
+        if not jobCE:
+            # Then we ended up with a site that doesn't exist?
+            logging.error("Job for non-existant site %s", job['location'])
+            return jdl
+
+        if self.submitWMSMode and len(job.get('possibleSites', [])) > 0:
+            strg = ','.join(map(str, job.get('possibleSites')))
+            jdl.append('+DESIRED_Sites = \"%s\"\n' % strg)
+        else:
+            jdl.append('+DESIRED_Sites = \"%s\"\n' % (jobCE))
+
+        if self.submitWMSMode and len(job.get('potentialSites', [])) > 0:
+            strg = ','.join(map(str, job.get('potentialSites')))
+            jdl.append('+ExtDESIRED_Sites = \"%s\"\n' % strg)
+        else:
+            jdl.append('+ExtDESIRED_Sites = \"%s\"\n' % (jobCE))
+
+        jdl.append('+WMAgent_RequestName = "%s"\n' % job['requestName'])
+        match = GROUP_NAME_RE.match(job['requestName'])
+        if match:
+            jdl.append('+CMSGroups = %s\n' % classad.quote(match.groups()[0]))
+        else:
+            jdl.append('+CMSGroups = undefined\n')
+
+        jdl.append('+WMAgent_SubTaskName = "%s"\n' % job['taskName'])
+        jdl.append('+CMS_JobType = "%s"\n' % job['taskType'])
+
+        # Handling for AWS, cloud and opportunistic resources
+        jdl.append('+AllowOpportunistic = %s\n' % job.get('allowOpportunistic', False))
+
+        # dataset info
+        if job.get('inputDataset'):
+            jdl.append('+DESIRED_CMSDataset = "%s"\n' % job['inputDataset'])
+        else:
+            jdl.append('+DESIRED_CMSDataset = undefined\n')
+        if job.get('inputDatasetLocations'):
+            jdl.append('+DESIRED_CMSDataLocations = "%s"\n' % ','.join(job['inputDatasetLocations']))
+        else:
+            jdl.append('+DESIRED_CMSDataLocations = undefined\n')
```

Comment:
It would be very, very, very useful to also provide the block being processed.  One of the bigger limitations now...